src/main/java/core/basesyntax/operation/SupplyHandler.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.service.ShopService;
+import core.basesyntax.transaction.FruitTransaction;
+
+public class SupplyHandler implements OperationHandler {
+    private ShopService shopService;
+
+    public SupplyHandler(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.SUPPLY) {
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+            shopService.addFruits(fruit, quantity);
+        } else {
+            throw new RuntimeException("Unsupported operation for SupplyHandler");
+        }

```

Comment:
This exception should never be thrown, because each handler should be used only for its specific operation. If this exception is thrown, it means there is a bug in the code where OperationHandler is used. So, it's better to remove this else block.